Library/Homebrew/cmd/update-bash.sh

```diff
@@ -36,26 +37,59 @@ git_init_if_necessary() {
     git config remote.origin.fetch "+refs/heads/*:refs/remotes/origin/*"
   fi
 
-  if git remote show origin -n | grep -q "mxcl/homebrew"
+  if [[ "$(git remote show origin -n)" = *"mxcl/homebrew"* ]]
   then
     git remote set-url origin https://github.com/Homebrew/homebrew.git
     git remote set-url --delete origin ".*mxcl\/homebrew.*"
   fi
 }
 
+rename_taps_dir_if_necessary() {
+  local tap_dir
+  for tap_dir in "$HOMEBREW_LIBRARY"/Taps/*
+  do
+    [[ -d "$tap_dir/.git" ]] || continue
+    local tap_dir_basename
+    tap_dir_basename="$(basename "$tap_dir")"
+    if [[ "$tap_dir_basename" = *"-"* ]]
+    then
+      # only replace the *last* dash: yes, tap filenames suck
+      local user
+      local repo
+      user="$(echo "${tap_dir_basename%-*}" | tr "[:upper:]" "[:lower:]")"
+      repo="$(echo "${tap_dir_basename:${#user}+1}" | tr "[:upper:]" "[:lower:]")"
+      mkdir -p "$HOMEBREW_LIBRARY/Taps/$user"
+      mv "$tap_dir", "$HOMEBREW_LIBRARY/Taps/$user/homebrew-$repo"
+
+      if [[ ${#${tap_dir_basename//[^\-]}} -gt 1 ]]
+      then
+        echo "Homebrew changed the structure of Taps like <someuser>/<sometap>. " >&2
+        echo "So you may need to rename $HOMEBREW_LIBRARY/Taps/$user/homebrew-$repo manually." >&2
+      fi
+    else
+      echo "Homebrew changed the structure of Taps like <someuser>/<sometap>. " >&2
+      echo "$tap_dir is incorrect name format. You may need to rename it like <someuser>/<sometap> manually." >&2
```

Comment:
I think it should be `to <someuser>/<sometap>` (instead of `like`) in both instances. The second line should probably better start with `$tap_dir is an invalid name. […]`.